lazyflow/operators/__init__.py

```diff
@@ -21,30 +19,53 @@
 # This information is also available on the ilastik web site at:
 # 		   http://ilastik.org/license/
 ###############################################################################
-import traceback, os, sys
+
 import logging
 
 logger = logging.getLogger(__name__)
 
-import lazyflow
-
 from lazyflow.graph import Operator
-from lazyflow.utility.helpers import itersubclasses
 
 from . import generic
 from . import filterOperators
 from . import classifierOperators
 from . import valueProviders
 from . import operators
 
-ops = itersubclasses(Operator)
+
+def _unique_subclasses_of(cls):
+    """Get unique sublcasses of a class.
+
+    Args:
+        cls: class with the `subclasses` attribute containing all it's subclasses
+
+    Returns:
+        {subclass_name: subclass} dict
+
+    Raises:
+        ImportError: duplicate subclass names
+    """
+    subs = {}
+    dupes = set()
+
+    for sub in cls.subclasses:
+        name = sub.__name__
+        if name in subs:
+            dupes.add(name)
+        subs[name] = sub
+
+    if dupes:
+        raise ImportError(f"some {cls.__name__} subclasses have identical names: {dupes}")
+
+    return subs
+
+
 logger.debug("Loading default Operators...")
-loaded = ""
-for i, o in enumerate(ops):
-    loaded += o.__name__ + " "
-    globals()[o.__name__] = o
-loaded += os.linesep
-logger.debug(loaded)
+subops = _unique_subclasses_of(Operator)
+globals().update(subops)
```

Comment:
since I have never looked a this code, is it, in short, just a fancy way of a star (`*`) import, filtering to only import operator sublcasses?
Maybe a comment saying this will safe some time whenever someone else looks at the code.

Reply:
>  is it, in short, just a fancy way of a star (*) import, filtering to only import operator sublcasses?

I guess it is a "recursive" start import, that is, it will import all (direct and indirect) subclasses that could be defined in subpackages.

I don't really like this approach. Do you think we need this? Maybe try to remove this and import operators explicitly?

Reply:
> I don't really like this approach. Do you think we need this? Maybe try to remove this and import operators explicitly?

I don't like it either and, honestly, wasn't aware that one can just import some random operator from `operators`. A quick grep gave me 88 matches for `from lazyflow.operators import`, so this pattern is in use. But this is nothing urgent. Maybe we can use your version for now and factor it out once there is time and these repos are merged?

Reply:
I would also be in favor of explicit imports, right now it breaks every known code autocompletion tool there is for `lazyflow.operators`

Reply:
okay I opened up an issue for this https://github.com/ilastik/ilastik/issues/2065

let's tackle this after the merge